File: HackerRank/Algorithms/Medium/M0011.cpp
/*
Problem Statement: https://www.hackerrank.com/challenges/bigger-is-greater/problem
*/

#include <iostream>
#include <string>
#include <algorithm>

using namespace std;

string biggerIsGreater(string w) {
    return (next_permutation(w.begin(), w.end())) ? w : "no answer";
}

int main() {
    int T;
    cin >> T;
    while (T--) {
        string w;
        cin >> w;
        cout << biggerIsGreater(w) << endl;
    }
    return 0;
}

File: HackerRank/Python/Medium/M0011.py
# Problem Statement: https://www.hackerrank.com/challenges/word-order/problem

from collections import OrderedDict

d = OrderedDict()

for _ in range(int(input())):
	w = input()
	d[w] = d.get(w, 0) + 1

print(len(d))
print(*d.values())